src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -538,7 +538,7 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
 
                 // Mark the consumed locals to indicate later drops are noops.
                 if let Place::Local(local) = *place {
-                    self.local_needs_drop[local] = None;
+                    self.local_needs_drop.remove(local.index());
```

Comment:
Hmm, I feel like this shouldn't happen unless we have a `Move`. Also, we can *probably* "just" remove `NEEDS_DROP` here from `self.temp_qualif[local]`, and check that later, instead of having a separate bitvec.